src/server/multi_svr.c

```diff
@@ -853,6 +853,42 @@ req_resc_update(int stream, pbs_list_head *ru_head, void *psvr)
 		send_command(stream, PS_RSC_UPDATE_ACK);
 }
 
+/**
+ * @brief open a multicast fd for peer servers corresponds to execvnode
+ * 
+ * @param[in] exec_vnode - exec vnode
+ * 
+ * @return int
+ * @retval -1 : for failure
+ */
+int
+open_ps_mtfd_for_execvnode(char *exec_vnode)
+{
+	int mtfd = -1;
+	char *chunk;
+	int rc = 0;
+	char *noden;
+	int nelem;
+	struct key_value_pair *pkvp;
+	struct pbsnode *pnode;
+
+	for (chunk = parse_plus_spec(exec_vnode, &rc);
+	     chunk && !rc; chunk = parse_plus_spec(NULL, &rc)) {
+
+		if (parse_node_resc(chunk, &noden, &nelem, &pkvp) == 0) {
+			if ((pnode = find_nodebyname(noden)) == NULL &&
+			    (pnode = find_alien_node(noden)) == NULL) {
+				close_streams(mtfd, rc);
+				mtfd = open_ps_mtfd();
+				return mtfd;
```

Comment:
Does this make it so that we broadcast to all if the node cache doesn't have info about a foreign node? Please add a comment above this block to explain the behavior.